storage/deposit/v1/store.go

```diff
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: BUSL-1.1
+//
+// Copyright (C) 2025, Berachain Foundation. All rights reserved.
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file of this repository and at www.mariadb.com/bsl11.
+//
+// ANY USE OF THE LICENSED WORK IN VIOLATION OF THIS LICENSE WILL AUTOMATICALLY
+// TERMINATE YOUR RIGHTS UNDER THIS LICENSE FOR THE CURRENT AND ALL OTHER
+// VERSIONS OF THE LICENSED WORK.
+//
+// THIS LICENSE DOES NOT GRANT YOU ANY RIGHT IN ANY TRADEMARK OR LOGO OF
+// LICENSOR OR ITS AFFILIATES (PROVIDED THAT YOU MAY USE A TRADEMARK OR LOGO OF
+// LICENSOR AS EXPRESSLY REQUIRED BY THIS LICENSE).
+//
+// TO THE EXTENT PERMITTED BY APPLICABLE LAW, THE LICENSED WORK IS PROVIDED ON
+// AN “AS IS” BASIS. LICENSOR HEREBY DISCLAIMS ALL WARRANTIES AND CONDITIONS,
+// EXPRESS OR IMPLIED, INCLUDING (WITHOUT LIMITATION) WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, NON-INFRINGEMENT, AND
+// TITLE.
+
+package deposit
+
+import (
+	"context"
+	"sync"
+
+	sdkcollections "cosmossdk.io/collections"
+	ctypes "github.com/berachain/beacon-kit/consensus-types/types"
+	"github.com/berachain/beacon-kit/errors"
+	"github.com/berachain/beacon-kit/log"
+	"github.com/berachain/beacon-kit/primitives/common"
+	"github.com/berachain/beacon-kit/storage"
+	depositstorecommon "github.com/berachain/beacon-kit/storage/deposit/common"
+	"github.com/berachain/beacon-kit/storage/encoding"
+	dbm "github.com/cosmos/cosmos-db"
+)
+
+const KeyDepositPrefix = "deposit"
+
+// KVStore is a simple KV store based implementation that assumes
+// the deposit indexes are tracked outside of the kv store.
+type KVStore struct {
+	store sdkcollections.Map[uint64, *ctypes.Deposit]
+
+	// closeFunc is a closure that closes the underlying database
+	// used by store to ensure that all writes are flushed to disk.
+	// We guarantee that closeFunc is called at maximum only once.
+	closeFunc CloseFunc
```

Comment:
moved mutex to generalStore since we only access this store via StoreManager interface